sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
     new ParseException(s"Found duplicate keys '$key'.", ctx)
   }
 
+  def formatForSetConfigurationUnExpectedError(ctx: SetConfigurationContext): Throwable = {
+    new ParseException(
+      s"""
+         |Expected format is 'SET', 'SET key', or 'SET key=value'. If you want to include
+         |special characters in key, or include semicolon in value, please use quotes,
+         |e.g., SET `ke y`=`v;alue`.
+       """.stripMargin.replaceAll("\n", " "), ctx)
+  }
+
+  def invalidPropertyKeyForSetQuotedConfigurationError(
+      keyCandidate: String, valueStr: String, ctx: SetQuotedConfigurationContext): Throwable = {
+    new ParseException(s"'$keyCandidate' is an invalid property key, please " +
+      s"use quotes, e.g. SET `$keyCandidate`=`$valueStr`", ctx)
+  }
+
+  def invalidPropertyValueForSetQuotedConfigurationError(
```

Comment:
invalidPropertyValueForSetConfigurationError

Reply:
It seems we should keep `invalidPropertyValueForSetQuotedConfigurationError`.

Reply:
Sounds good. We can emphasize the property key/value should be quoted.